### ISSUE TITLE
Pa11y integration with already working Puppeteer + Lighthouse fails on pages with auth

### ISSUE THREAD

**Issue by ray-kim-12** [pa11y/pa11y#410]
I had a fully working Puppeteer + Lighthouse and inserted Pa11y for its (what I think) superior accessibility monitoring ability. It works on publicly available URLs and locally available URLs, but pa11y does NOT work on already-logged in (via Puppeteer) pages behind auth.

## Expected behaviour
Because I'm already logged in successfully via Puppeteer, I expect to be able to generate a Pa11y report on the current page (the callback page upon successful login, the one behind auth). 

## Actual behaviour
However, I am hit with a sign in to access this page (the one behind auth)

```
  try {
    await page.goto(LOGIN_PAGE) // at login page
    await generate_pally_html(LOGIN_PAGE) // successfully generates pally report on this
    await login(page, browser); // login
    console.log('after 1st attempt to login: should be back to localhost:3000 ?==', page.url())
  } catch (error) {
    console.log('attempting to go to login page: ', error);
  }

  while (page.url() != HOME_PAGE) { 
    console.log('unsuccessful login; trying to login again')
    try {
      await login(page, browser);
      console.log('should be logged in, back to 3000: ', page.url())
    } catch (err) {
      console.log('attempted to login failed with message: ' + error)
    }
  }

  await page.goto(BUILDER_PAGE); 
  while (page.url() !== BUILDER_PAGE) {
    try {
      await page.goto(BUILDER_PAGE);
      console.log('should be at builder: ', page.url())
    } catch (err) {
      console.log('failed to go to /mapping/builder with: ', err)
      console.log('at this url instead: ', page.url())
    }
  }
  // Currently at page behind auth, 

  await generate_pally_html(BUILDER_PAGE) // generates a pa11y report on page that requests me to log in again, even though I had already logged in 
  await builder_lighthouse(page, browser); // successfully generates lighthouse report on that same page
```

## Environment:
pa11y --environment
Pa11y:      5.0.4
Node.js:    10.1.0
npm:        5.6.0
OS:         17.5.0 (darwin)

I recently just discovered pa11y, but I am already in awe of the simplicity and the immense power of this technology. Kudos to the team !

**Comment by rowanmanning** [comment]
Hi @ray-kim-12, thanks for the feedback :) in order to help debug it'd be useful if we could see what the `generate_pa11y_html` function does

**Comment by hollsk** [comment]
Hi @ray-kim-12 - do you still need help with this issue? 😄

**Comment by josebolos** [comment]
There's been no activity for a while, so I'm closing the issue. Please feel free to reopen it if you're still experiencing this! 👍
